bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -652,6 +652,7 @@ void DWARFRewriter::updateDebugInfo() {
         auto LegacyRangesSectionWriterByCU =
             std::make_unique<DebugRangesSectionWriter>();
         LegacyRangesSectionWriterByCU->initSection(CU);
+        LegacyRangesSectionWriterByCU->clearBuffer();
```

Comment:
Since we have each ranges writer its own instance, they all act the same and we still want the empty buffer at the beginning of debug_ranges which is in the main ranges writer. We could avoid creating empty ranges by passing some sort of flag into the constructor or some similar solution.

Reply:
Sure give it a shot, or remove init code entirely and for main one write out empty list explicilty?